pkg/controller/migmigration/stage.go

```diff
@@ -91,6 +90,46 @@ func BuildStagePods(labels map[string]string,
 	return stagePods
 }
 
+// get the running application pods with attached PVC
+func GetApplicationPodsWithStageLabels(labels map[string]string, pvcMapping map[k8sclient.ObjectKey]migapi.PV, list *[]corev1.Pod) []corev1.Pod {
+	applicationPodsWithStageLabel := []corev1.Pod{}
+	for _, pod := range *list {
+		foundVolume := true
+		excludePVC := ""
+		for _, volume := range pod.Spec.Volumes {
+			if volume.PersistentVolumeClaim == nil {
+				continue
+			}
+			claimKey := k8sclient.ObjectKey{
+				Name:      volume.PersistentVolumeClaim.ClaimName,
+				Namespace: pod.Namespace,
+			}
+			pv, found := pvcMapping[claimKey]
+			if !found {
+				continue
+			}
+			if pv.Selection.Action != migapi.PvCopyAction ||
+				pv.Selection.CopyMethod != migapi.PvFilesystemCopyMethod {
+				excludePVC += volume.Name + ";"
+				continue
+			}
+			foundVolume = false
+		}
+		if foundVolume {
+			continue
+		}
+		if excludePVC != "" {
+			pod.Labels[migapi.ExcludePVCPodLabel] = excludePVC
```

Comment:
Considering the comments above, this would be something like the following instead:
```
pod.Annotations[migapi.ExcludePVCPodAnnotation] = strings.Join(excludePVC, ",")
```